torchci/lib/bot/autoLabelBot.ts

```diff
@@ -1,10 +1,71 @@
 import { Context, Probot } from "probot";
 
-const regexToLabel: [RegExp, string][] = [
+const titleRegexToLabel: [RegExp, string][] = [
   [/rocm/gi, "module: rocm"],
   [/DISABLED\s+test.*\(.*\)/g, "skipped"],
 ];
 
+const filenameRegexToReleaseCategory: [RegExp, string][] = [
+  // dataloader_frontend => does that correlate best with release notes: AO frontend?
+  [/torch\/utils\/data/gi, "module: dataloader"],
+  [/test_data(loader|pipe)/gi, "module: dataloader"],
+  // cpp_frontend => not a real category either
+  [/torch\/(csrc|cpp)\/api/gi, "module: cpp"],
+  // distributed + its flavors
+  [/c10d/gi, "release notes: distributed (c10d)"],
+  [/distributed.*sharded/gi, "release notes: distributed (sharded)"],
+  [/distributed.*ddp/gi, "release notes: distributed (ddp)"],
+  [/distributed.*pipeline/gi, "release notes: distributed (pipeline)"],
+  [/distributed.*fsdp/gi, "release notes: distributed (fsdp)"],
+  [/distributed.*rpc/gi, "release notes: distributed (rpc)"],
+  // vulkan
+  [/vulkan/gi, "release notes: vulkan"],
+  // foreach_frontend
+  [/foreach/gi, "release notes: foreach_frontend"],
+  // onnx
+  [/onnx/gi, "release notes: onnx"],
+  // fx
+  [/torch\/fx/gi, "release notes: fx"],
+  [/test_fx/gi, "release notes: fx"],
+  // ao
+  [/(torch|test)\/ao/gi, "release notes: AO frontend"],
+  // quantization
+  [/(torch|test)\/quantization/gi, "release notes: quantization"],
+  [/aten\/src\/ATen\/native\/quantized/gi, "release notes: quantization"],
+  [/torch\/nn\/quantiz(ed|able)/gi, "release notes: quantization"],
+  // package => is this supposed to be package/deploy?
+  [/(torch|test)\/package/gi, "release notes: package/deploy"],
+  // mobile
+  [/torch\/csrc\/jit\/mobile/gi, "release notes: mobile"],
+  [/aten\/src\/ATen\/native\/metal/gi, "release notes: mobile"],
+  [/test\/mobile/gi, "release notes: mobile"],
+  [/torch\/backends\/_nnapi\//gi, "release notes: mobile"],
+  [/test\/test_nnapi.py/gi, "release notes: mobile"],
+  // linalg_frontend
+  [/aten\/src\/ATen\/native\/LinearAlgebra.cpp/gi, "release notes: linalg_frontend"],
+  [/test\/test_linalg.py/gi, "release notes: linalg_frontend"],
+  [/torch\/linalg/gi, "release notes: linalg_frontend"],
+  // sparse_frontend => also did not exist
+  [/aten\/src\/ATen\/native\/sparse/gi, "module: sparse"],
+  [/torch\/sparse/gi, "module: sparse"],
+  [/torch\/_masked\/__init__.py/gi, "module: sparse"],
+  // autograd_frontend => also did not exist
+  [/tools\/autograd/gi, "module: autograd"],
+  // nn_frontend => also did not exist
+  [/test\/test_nn.py/gi, "module: nn"],
+  [/test\/test_module.py/gi, "module: nn"],
+  [/tools\/nn\/modules/gi, "module: nn"],
+  [/tools\/nn\/functional.py/gi, "module: nn"],
```

Comment:
is using the module prefix correct here?

Reply:
We might want to use the "release notes: "

FYI: Talking with @albanD, the original reason for having the extra set was so that we sometimes wanted to them to be independent of the "module: " labels. Module owners might want to have more fine-grained labels for the purposes of release notes (for example distributed already does, and JIT might want to add)

Reply:
Oh, to clarify, the reason these aren't "release notes" is cuz the release notes labels for these don't exist.

Reply:
Sounds like the action here would be to create these nonexistent labels?

Reply:
We do have these actually: `release notes: nn` does exist.